File: LocalAIAgent.API/Infrastructure/Models/AiSettings.cs
namespace LocalAIAgent.API.Infrastructure.Models
{
    public class AiSettings
    {
        public int Id { get; set; }
        public required string ModelId { get; set; } = "gemma-3-27b-it-qat";
        public string ApiKey { get; set; } = string.Empty;
        public required string EndpointUrl { get; set; } = "http://localhost:1234/v1/";
        public decimal Temperature { get; set; } = 0.2m;
        public decimal TopP { get; set; } = 1.0m;
        public decimal FrequencyPenalty { get; set; } = 1.0m;
        public decimal PresencePenalty { get; set; } = 1.0m;
        public int UserPreferencesId { get; set; }

        public required virtual UserPreferences UserPreferences { get; set; }
    }
}

File: LocalAIAgent.WebUI/src/domain/AISettings.ts
export default class AISettings {
    modelId: string;
    apikey: string;
    endpointUrl: string;
    temperature: number;
    topP: number;
    frequencyPenalty: number;
    presencePenalty: number;

    constructor(
        modelId: string = 'gemma-3-27b-it-qat', 
        apikey: string = '', 
        endpointUrl: string = 'http://localhost:1234/v1/', 
        temperature: number = 0.2,
        topP: number = 1.0,
        frequencyPenalty: number = 1.0,
        presencePenalty: number = 1.0) {
        this.modelId = modelId;
        this.apikey = apikey;
        this.endpointUrl = endpointUrl;
        this.temperature = temperature;
        this.topP = topP;
        this.frequencyPenalty = frequencyPenalty;
        this.presencePenalty = presencePenalty;
    }
}
